Tools/GNUMake/Make.rules

```diff
@@ -125,13 +138,39 @@ install_headers:
 	done
 	@install -c -m 644 $(srcTempDir)/AMReX_Config.H $(amrexIncludeDir)
 
+uninstall_headers:
+	@echo Uninstall headers at $(amrexIncludeDir) ...
+	@for d in $(Pdirs); do \
+	  for h in $(CEXE_headers) $(cEXE_headers); do \
+	    f=Src/$$d/$$h; \
+	    if test -f "$$f"; then \
+	       $(SILENT) $(RM) $(amrexIncludeDir)/$$h; \
+	    fi; \
+	  done; \
+	done
+	@$(SILENT) $(RM) $(amrexIncludeDir)/AMReX_Config.H;
+	@if [ -d $(amrexIncludeDir) ]; then \
+	    $(SILENT) rmdir -p --ignore-fail-on-non-empty $(amrexIncludeDir); \
+	fi;
+
 install_fortran_modules:
 	@echo Install Fortran modules at $(amrexIncludeDir) ...
 	@if [ ! -d $(amrexIncludeDir) ]; then mkdir -p $(amrexIncludeDir); fi
 	@for m in $(amrexFortModules); do \
 	    install -c -m 644 "$$m" $(amrexIncludeDir); \
 	done
 
+uninstall_fortran_modules:
+	@echo Uninstall Fortran modules at $(amrexIncludeDir) and $(fmoddir) ...
+	@for m in $(amrexFortModules); do \
+	    $(SILENT) $(RM) $$m; \
+	    $(SILENT) $(RM) $(amrexIncludeDir)/*.mod; \
```

Comment:
`$$m` instead of `*.mod`?

Reply:
`*.mod` is removed by distclean. `$$m` should only remove the `.mod` files we wrote

Reply:
I thought `amrexIncludeDir` could be something like `/usr/local` and therefore we don't want to `rm /usr/local/*.mod`.  :thinking:

Reply:
OK.  I think we want `$(amrex/IncludeDir)/$(notdir $m)`.

Reply:
`amrexIncludeDir` is hardcoded in Make.rules to be `tmp_build_dir`. The original issue was if `AMREX_INSTALL_DIR` gets set to a non-empty directory:
`$(SILENT) $(RM) -r $(AMREX_INSTALL_DIR)`

Reply:
> amrexIncludeDir is hardcoded in Make.rules to be tmp_build_dir

Can you point me to where that is?  I only see
```
amrexIncludeDir = $(AMREX_INSTALL_DIR)/include
```
where
```
AMREX_INSTALL_DIR ?= tmp_build_dir
```
`?=` means the user could set it to anything.

I understand the original issue. But we should not do `rm /usr/local/*.mod` either.

Reply:
I see what you mean now. If we don't want to do `rm /usr/local/*.mod`, then should I revert the distclean commit suggestion? Since that's effectively what using distclean instead of something more complicated would do...
https://github.com/AMReX-Codes/amrex/pull/2471/commits/c3a49cca0b32a8bfaaeb47fd789f1694097428d7

Reply:
In `configure.py`,
```
    parser.add_argument("--prefix",
                        help="Install libamrex, headers and Fortran modules in PREFIX directory [default=tmp_install_dir]",
                        default="tmp_install_dir")
# ...
    f = open("GNUmakefile","w")
    f.write("AMREX_INSTALL_DIR = " + args.prefix.strip() + "\n")
```

So `AMREX_INSTALL_DIR` is either `tmp_intall_dir` by default or the directory provided by the user with `--prefix`.  Both are different from `tmp_build_dir`.